prow/cmd/branchprotector/protect_test.go

```diff
@@ -717,3 +717,61 @@ func fixup(r *requirements) {
 		sort.Strings(*restr.Users)
 	}
 }
+
+func TestIgnoreArchivedRepos(t *testing.T) {
```

Comment:
An alternative here would be to add something in `GetRepos()` in `fakeClient` to where `for i, rr := range r { if strings.Contains(rr, "archived")` { rr.Archived = true; r[i] = rr } }`

Then you could test this in `TestProtect` by setting `branches: []string{"blah/repo=master", "blah/archived-repo=master"}` and ensuring we only protected `blah/repo`

This is perhaps less magical though so also good

Reply:
Yeah I wanted to not complicate that setup any more than it had already been, this seemed more straightforward